### PR TITLE
test(e2e): add suite for i18n plugin

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* adds e2e tests for i18n that test the feature as part of the application through various veritcal journeys
* enhances the settings table with better accessibility labels
* if you try to go to a locale that does not exist, we redirect you to the default in the edit & list views
* fixes situations where the locale injected UI was not correctly disabled
* fixes issue with UID field where the default value kept overwrite what ever we wanted the regeneration to be

### CODE REVIEW

**Comment by joshuaellis** [comment]
@MarionLemaire I have quite a few to write, but i thought i'd give you visibility to what i was thinking for this first iteration. You can also see the variety in content-type abilities too.

File: packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/tests/UID.test.tsx [hunk 1]
```diff
@@ -72,17 +72,17 @@ describe('UIDInput', () => {
     expect(screen.getByRole('textbox', { name: 'Label' })).toHaveValue('regenerated');
   });
 
-  test('If the field is required and the value is empty it should automatically fill it', async () => {
+  test.only('If the field is required and the value is empty it should automatically fill it', async () => {
```
Comment (jhoward1994) [inline packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/tests/UID.test.tsx:75]:
```suggestion
  test('If the field is required and the value is empty it should automatically fill it', async () => {
```

File: packages/plugins/i18n/admin/src/translations/en.json [hunk 2]
```diff
@@ -14,11 +14,11 @@
   "Field.localized": "This value is unique for the selected locale",
   "Field.not-localized": "This value is common to all locales",
   "Settings.list.actions.add": "Add new locale",
-  "Settings.list.actions.delete": "Delete a locale",
+  "Settings.list.actions.delete": "Delete {name} locale",
   "Settings.list.actions.deleteAdditionalInfos": "This will delete the active locale versions <em>(from Internationalization)</em>",
   "Settings.list.actions.publishAdditionalInfos": "This will publish the active locale versions <em>(from Internationalization)</em>",
   "Settings.list.actions.unpublishAdditionalInfos": "This will unpublish the active locale versions <em>(from Internationalization)</em>",
-  "Settings.list.actions.edit": "Edit a locale",
+  "Settings.list.actions.edit": "Edit {name} locale",
```
Comment (jhoward1994) [inline packages/plugins/i18n/admin/src/translations/en.json:21]: A bit nit picky but "Edit locale Fr" would read better to me than "Edit Fr
locale". wdyt? 🤷‍♂️

File: e2e/tests/i18n/editview.spec.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    await expect(page.getByRole('heading', { name: 'Products' })).toBeVisible();
+
+    /**
+     * Assert we're on the english locale and our document exists
+     */
+    await expect(page.getByRole('combobox', { name: 'Select a locale' })).toHaveText(
+      'English (en)'
+    );
+    await expect(
+      page.getByRole('row', { name: 'Nike Mens 23/24 Away Stadium Jersey' })
```
Comment (jhoward1994) [inline e2e/tests/i18n/editview.spec.ts:36]: 😆 the files I wrote are becoming such a let down, I need to up my game.
Football is life!

File: packages/plugins/i18n/admin/src/translations/en.json (hunk 2 above)
Comment (joshuaellis) [reply packages/plugins/i18n/admin/src/translations/en.json:21]: You've made me doubt myself now lol, it'll normally read "Edit French (fr) locale" that's fine right? 😅

Comment (jhoward1994) [reply packages/plugins/i18n/admin/src/translations/en.json:21]: Yeah your way is better 😆. Sorry I should've looked at the tests before commenting

File: e2e/tests/i18n/settings.spec.ts [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+     * Because this is technically a new entry, we should assert that the document actions are disabled.
+     */
+    await expect(page.getByRole('button', { name: 'Publish' })).toBeDisabled();
+    await expect(page.getByRole('button', { name: 'Save' })).toBeDisabled();
+    await expect(page.getByRole('button', { name: 'More document actions' })).toBeDisabled();
+    await expect(page.getByRole('button', { name: 'More actions' })).toBeEnabled();
+    await page.getByRole('button', { name: 'More actions' }).click();
+    await expect(page.getByRole('menuitem', { name: 'Edit the model' })).toBeEnabled();
+    await expect(page.getByRole('menuitem', { name: 'Configure the view' })).toBeEnabled();
+    await expect(page.getByRole('menuitem', { name: 'Delete locale' })).toBeDisabled();
```
Comment (jhoward1994) [inline e2e/tests/i18n/settings.spec.ts:72]: I wonder if 'Delete locale' in the CM could be confusing and make people think
they're removing the locale completely?

**Comment by joshuaellis** [comment]
im gonna open a new PR.

File: e2e/tests/i18n/settings.spec.ts (hunk 4 above)
Comment (lucasboilly) [reply e2e/tests/i18n/settings.spec.ts:72]: Hey @jhoward1994 it could, indeed. I'm gonna speak about this with Paridhi :)

**Pull request closed without merging**